vm/loader/src/paravisor.rs

```diff
@@ -321,6 +321,13 @@ where
 
     offset += HV_PAGE_SIZE;
 
+    // Reserve space for the VTL2 reserved region.
+    let reserved_region_size = PARAVISOR_RESERVED_VTL2_PAGE_COUNT_MAX * HV_PAGE_SIZE;
```

Comment:
We're reserving space on all architectures even though only SNP uses it? Is that just for simplicity?

Reply:
On x64 only. But yes, it makes things simpler. I'm not too concerned about a losing a few pages, but if need be we could modify the loader so that we get these pages back, if we need to.

Reply:
it just makes layout calculation more complicated. Maybe I'll fix all of it again later, but I don't think it's worth doing now.

Note that these pages were already reserved everywhere before.

Reply:
Were they? I thought for "VSM" we didn't include the secrets page, vmsa, or cpuid pages:

```
pub const PARAVISOR_UNMEASURED_VTL2_CONFIG_REGION_PAGE_COUNT_VSM: u64 =
    PARAVISOR_UNMEASURED_VTL2_CONFIG_REGION_PAGE_COUNT_COMMON;
/// Size in pages for AMD SEV-SNP isolation.
pub const PARAVISOR_UNMEASURED_VTL2_CONFIG_REGION_PAGE_COUNT_SNP: u64 =
    PARAVISOR_UNMEASURED_VTL2_CONFIG_REGION_PAGE_COUNT_COMMON
        + PARAVISOR_CONFIG_SECRETS_SIZE_PAGES
        + PARAVISOR_CONFIG_CPUID_SIZE_PAGES
        + PARAVISOR_CONFIG_VMSA_SIZE_PAGES;
```

Reply:
we used max always, not VSM.

Reply:
```rust

    let parameter_region_size = PARAVISOR_VTL2_CONFIG_REGION_PAGE_COUNT_MAX * HV_PAGE_SIZE;
 ```

Reply:
We should probably file an issue to track making all of these calculations work per-isolation type so we don't waste pages.